File: spec_backend_com_python/aulas/exercicios/27_alunos_media_01/aluno.py
class Aluno:
    def __init__(self):
        self.nome = ""
        self.idade = 0
        self.matricula = 0
        self.media_notas = 0.0

    def cadastrar_dados(self, nome, idade, matricula):
        self.nome = nome
        self.idade = idade
        self.matricula = matricula

    def cadastrar_media_notas(self, media):
        self.media_notas = media

    def exibir_informacoes(self):
        print(f"Nome: {self.nome}")
        print(f"Idade: {self.idade}")
        print(f"Matrícula: {self.matricula}")
        print(f"Média de Notas: {self.media_notas}")

File: spec_backend_com_java_e_springboot/resolvidos/exercicio_27/alunos_media_03/src/Aluno.java
public class Aluno {
    // Atributos (variáveis de instância)
    private String nome;
    private int idade;
    
    
    private int matricula;
    private double prova1;
    private double prova2;

    // Método para cadastrar os dados básicos
    public void cadastrarDados(String Nome, int Idade, int Matricula) {
        this.nome = Nome;
        this.idade = Idade;
        this.matricula = Matricula;
    }

    // Método para cadastrar as notas das provas
    public void setNotas(double notaProva1, double notaProva2){
        this.prova1 = notaProva1;
        this.prova2 = notaProva2;
    }

    // Método para calcular a média de notas
    public  double calcularMedia() {
        return (prova1 + prova2)/2;
    }

    // Método para exibir todas as informações
    public void exibirInformacoes() {
        System.out.println("=== DADOS DO ALUNO ===");
        System.out.println("Nome: " + nome);
        System.out.println("Idade: " + idade);
        System.out.println("Matrícula: " + matricula);
        System.out.println("Média de notas: " + calcularMedia());
    }
}